### ISSUE TITLE
Org-roam buffer fails to load

### ISSUE THREAD

**Issue by tpeacock19** [jethrokuan/org-roam#135]
### Description
Org-roam buffer throws an error upon loading.
#### Steps to Reproduce
1. Load Emacs
2. Run `org-roam--build-cache-async`
3. Run `org-roam-find-file` and select a note
4. Run `org-roam`

#### Expected Results

Org-roam buffer should load with backlinks.
#### Actual Results
Org-roam buffer loads blank and I receive the following error:
```elisp 
Error in post-command-hook (org-roam--maybe-update-buffer): (wrong-type-argument arrayp nil)
```
This error began with the commit referenced below and persists through the most recent on Develop

The buffer will show backlinks to notes opened with `org-roam-find-file` _after_ the org-roam buffer is loaded; however, it does not show for the original note. Additionally, it does not seem to share the same formatting as my `org-mode` buffers as it has prior to this commit.
 
### Versions
- Emacs (`C-h v emacs-version`): v28.0.50
- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/bdc13cd6bff3a4869ace8601bbbb516c6851b3b4

**Comment by jethrokuan** [comment]
@tpeacock19 When you say error persists through most recent `develop`, did you mean you've also checked that https://github.com/jethrokuan/org-roam/pull/128 (merged after the commit hash you mentioned) didn't resolve the issue?

> The buffer will show backlinks to notes opened with org-roam-find-file after the org-roam buffer is loaded; however, it does not show for the original note.

That is correct, this has always been the expected behaviour atm, I could force a call to `org-roam--maybe-update-buffer` when `org-roam` is called.

> Additionally, it does not seem to share the same formatting as my org-mode buffers as it has prior to this commit.

But this is odd, the formatting hasn't changed recently. What are you seeing that is weird to you?

**Comment by tpeacock19** [comment]
I'm seeing this issue when using develop up to this commit: 63754d1ccd14f477396f8ee5342fc14e2f75adee this should include that merge, right?

The formatting looks like this:
![2020-02-19-220047_915x1007_scrot](https://user-images.githubusercontent.com/6407145/74902629-d8a85680-5363-11ea-85cf-3aa2de506b37.png)

When it should show this (my custom fonts & org-bullets are loaded):
![2020-02-19-220133_911x1002_scrot](https://user-images.githubusercontent.com/6407145/74902652-e8279f80-5363-11ea-9cfa-ebdf2b707126.png)

**Comment by tpeacock19** [comment]
Hm, It seems like this has resolved itself with the commit 0346d3b16c42e6a05e34daa2fb91dddf8cfe714b posted a bit after I submitted this issue.  

![2020-02-19-221200_910x1004_scrot](https://user-images.githubusercontent.com/6407145/74903131-5456d300-5365-11ea-9378-d16e7e1f431f.png)


I'm not sure what the issue was but I can close this out if you wish.

**Comment by jethrokuan** [comment]
Oh... Damn these things are so hard to trace back. I guess `org-roam-file-p` failing caused the `org-roam` buffer update to fail, which means your hooks didn't complete running and resulting in the unstyled formatting you see.

**Comment by Cumol** [comment]
Maybe related but I am not getting an empty *org-roam* buffer. Completely blank, even though I crated the cache etc.

**Comment by teesloane** [comment]
@Cumol I'm getting an empty buffer too. Any luck fixing this?

<img width="1400" alt="Screen Shot 2020-02-23 at 5 59 06 PM" src="https://user-images.githubusercontent.com/12987958/75121862-355a8880-5666-11ea-8f3d-44fe39c81292.png">

Here's my configuration of org-roam in my config (using doom-emacs)

```elisp
(defun tees/org-roam-title-private (title)
  (let ((timestamp (format-time-string "%Y%m%d%H%M%S" (current-time)))
        (slug (org-roam--title-to-slug title)))
    (format "private-%s_%s" timestamp slug)))

(use-package! org-roam
  :after org
  :config
  (setq-default
   org-roam-directory "~/Dropbox/wiki"
   org-roam-link-title-format "R:%s"
   org-roam-templates
   (list
    (list "default" (list :file #'org-roam--file-name-timestamp-title :content "
#+TITLE: ${title}
#+SLUG: ${slug}
"))
    (list "private" (list :file #'tees/org-roam-title-private
                               :content "#+TITLE: ${title}"))))


  (org-roam--build-cache-async)

  (map! :map org-mode-map
        :localleader
        (:prefix ("R" . "Org Roam")
          "o"   #'org-roam
          "f"   #'org-roam-find-file
          "g"   #'org-roam-show-graph
          "i"   #'org-roam-insert)))
```

Very cool package. Looking forward to using it.

**Comment by jethrokuan** [comment]
it doesn't seem like `org-roam-mode` is turned on by default in your config, which is probably why.

**Comment by teesloane** [comment]
That was the case @jethrokuan ; thank you!

**Comment by Cumol** [comment]
These are my settings and I am not getting any backlinks

```
(use-package org-roam
      :after org
      :hook 
      (after-init . org-roam-mode)
      :straight (:host github :repo "jethrokuan/org-roam" :branch "develop")
      :custom
      (org-roam-directory "~/cloud/phd/notes")
      :bind (:map org-roam-mode-map
              (("C-c n l" . org-roam)
               ("C-c n f" . org-roam-find-file)
               ("C-c n g" . org-roam-show-graph))
              :map org-mode-map
              (("C-c n i" . org-roam-insert))))
```

It doesn't seem like org-roam-mode is on by default, even though I have the after-init hook

Any idea what is happening?

**Comment by Cumol** [comment]
Pulled the latest develop from straight. org-roam-mode is not activated by default but if I do:
 `org-roam-mode` -> `org-roam--build-cache-async` -> `org-roam-find-file` -> `org-roam
`
It works.

**Comment by asymmetric** [comment]
I have the same issue as @Cumol, and following the steps in the comment above is not enough.

There seems to be a weird interaction between how one switches buffers, whether the `org-roam-buffer` is open, for it to be displayed at all.

* Emacs: v26.3
* Doom: v2.0.9
* org-roam: e35769329783713f9a49b5c0285c941f68b3b2dd

**Comment by asymmetric** [comment]
I've tried removing the `org-roam.db` and letting org-roam recreate it (happens automatically AFAICT), but encountered the same issues.

**Comment by jethrokuan** [comment]
I see that your `org-roam` buffer is stuck in fundamental mode, so something probably failed while trying to initialize the buffer. Could you check your messages, or turn on `debug-on-error`?

**Comment by asymmetric** [comment]
The problem seems to have been solved in either latest Doom or latest org-roam, so all good :)

**Comment by neilyio** [comment]
I also just had this problem after a new Emacs install. The `*org-roam*` buffer would be blank and in fundamental mode. No matter how often I called `C-u M-x- org-roam-db-build-cache`, I couldn't get it to show anything, not even "no backlinks".  Using `org-roam-find-file` would not show the `#+TITLE` of each file, but instead the datestamp file name, e.g. `20200406155035.org`.

I'm using Spacemacs, and I had copy pasted the `~/.emacs.d/private/org-roam/packages.el` from [the documentation](https://org-roam.readthedocs.io/en/master/installation/). I replaced `/path/to/org-files/"`with `/Users/neil/Documents/notes/`.

Not knowing what else to do to get org-roam working, and tried removing the trailing slash on my path name, changing `/Users/neil/Documents/notes/` to `/Users/neil/Documents/notes`. I restarted Emacs, called `C-u M-x- org-roam-db-build-cache`, and it worked!

This was my fourth time re-starting Emacs and calling `C-u M-x- org-roam-db-build-cache`, so maybe that was just a lucky number of restarts. But the trailing slash was the only change I made between attempts.

**Comment by jethrokuan** [comment]
@neilyio you were likely on an old (but recent) version of org-roam, this was fixed in #735 .

**Comment by detrout** [comment]
I found this report because I was also experiencing
```Error in post-command-hook (org-roam--maybe-update-buffer): (wrong-type-argument arrayp nil)```
but it turned out to be a configuration error. I'm leaving this comment here in case someone else has the same problem as I did.

What happened was I thought I would test org-roam in an isolated subtree of my org hierarchy by setting ```org-roam-directory``` and ```org-roam-db-location``` in ```.dir-locals.el```.

I then added ```(add-hook 'after-init-hook 'org-roam-mode)``` to my init.el.

The documentation mentions you need to set ```org-roam-directory``` before adding the init-hook, but I missed that. The error went away when I set an ```(setq org-roam-directory ...)``` before calling ```(add-hook 'after-init-hook 'org-roam-mode)```

Setting the ```org-roam-directory``` in a .dir-locals.el variable appears to happens after the init.el file is processed.
